src/assert.bash

```diff
@@ -205,17 +212,25 @@ assert_output() {
 
   # Arguments.
   local expected
-  (( $# == 0 )) && expected="$(cat -)" || expected="$1"
+  if (( use_stdin )); then
+    expected="$(cat -)"
+  else
+    expected="$1"
+  fi
 
   # Matching.
-  if (( is_mode_regexp )); then
+  if (( is_mode_nonempty )); then
+    if [ -z "$output" ]; then
+      echo 'expected non-empty output, but output was empty' \
+      | batslib_decorate 'no output' \
```

Comment:
No strong preference. I just use the formatter that comes built-in with vim, which had them not indented.